File: VoteUp/src/main/java/net/shoal/sir/voteup/api/VoteUpAPI.java
package net.shoal.sir.voteup.api;

import net.shoal.sir.voteup.VoteUp;
import net.shoal.sir.voteup.config.CacheManager;
import net.shoal.sir.voteup.config.GuiManager;
import net.shoal.sir.voteup.config.VoteManager;
import net.shoal.sir.voteup.config.VoteUpConfig;

public class VoteUpAPI {
    public static final VoteManager VOTE_MANAGER;
    public static final GuiManager GUI_MANAGER;
    public static final CacheManager CACHE_MANAGER;
    public static final VoteUpConfig CONFIG;

    static {
        VOTE_MANAGER = new VoteManager();
        GUI_MANAGER = new GuiManager();
        CACHE_MANAGER = new CacheManager();
        CONFIG = (VoteUpConfig) VoteUp.getInstance().pConfig;
    }
}

File: VoteUp/src/main/java/net/shoal/sir/voteup/api/VoteUpAPI.kt
package net.shoal.sir.voteup.api

import net.shoal.sir.voteup.config.CacheManager
import net.shoal.sir.voteup.config.GuiManager
import net.shoal.sir.voteup.config.VoteManager

object VoteUpAPI {
    var SOUND: SoundUtil? = null
    var VOTE_MANAGER: VoteManager? = null
    var GUI_MANAGER: GuiManager? = null
    var CACHE_MANAGER: CacheManager? = null

    init {
        SOUND = SoundUtil()
        VOTE_MANAGER = VoteManager()
        GUI_MANAGER = GuiManager()
        CACHE_MANAGER = CacheManager()
    }
}